src/perform_step/extrapolation_perform_step.jl

```diff
@@ -955,12 +955,22 @@ function perform_step!(integrator, cache::ImplicitDeuflhardExtrapolationCache, r
         # Update cache.T
         j_int = 2 * subdividing_sequence[n_curr + 1]
         dt_int = dt / j_int # Stepsize of the new internal discretisation
+        jacobian2W!(W, integrator.f.mass_matrix, dt_int, J, false)
+        integrator.destats.nw += 1
         @.. u_temp2 = uprev
-        @.. u_temp1 = u_temp2 + dt_int * fsalfirst # Euler starting step
+        @.. linsolve_tmp = dt_int*fsalfirst
+        cache.linsolve(vec(k), W, vec(linsolve_tmp), !repeat_step)
+        integrator.destats.nsolve += 1
+        @.. k = -k
```

Comment:
what's this?

Reply:
This is for the T[n_curr + 1] approximation using h = sequence[n_curr+1]. The mentioned code solves the semi-implicit Euler equation for the starting step. Then rest iteration for semi-implicit midpoint follows. This is similar to https://github.com/SciML/OrdinaryDiffEq.jl/blob/master/src/perform_step/extrapolation_perform_step.jl#L1646-L1666. Previously T[n_curr + 1] approximation was being done using explicit-midpoint, which I guess got copied from extrapolation methods adaptivity. Since we are now dealing with implicit methods, we should use the semi-implicit formula for getting T[n_curr + 1] as mentioned in the attached link [here](https://github.com/SciML/OrdinaryDiffEq.jl/blob/master/src/perform_step/extrapolation_perform_step.jl#L1646-L1666).